server/src/test/java/org/elasticsearch/action/search/AbstractSearchAsyncActionTests.java

```diff
@@ -82,7 +82,9 @@ private AbstractSearchAsyncAction<SearchPhaseResult> createAction(
             null,
             request,
             listener,
-            new GroupShardsIterator<>(Collections.singletonList(new SearchShardIterator(null, null, Collections.emptyList(), null))),
+            new GroupShardsIterator<>(
+                Collections.singletonList(new SearchShardIterator(null, new ShardId("index", "_na", 0), Collections.emptyList(), null))
```

Comment:
Why do we no longer accept null here? Do we have risks of NPEs elsewhere?

Reply:
We never accepted `null` for the `ShardId` past the constructor. E.g. the equals method on the class assumes it's non-null.